### PR TITLE
IA-2398 Add a `Teams` filter on users page

### PR DESCRIPTION
**Pull request by kemar**
Add a `Team` filter on users page.

Related JIRA tickets : [IA-2398](https://bluesquare.atlassian.net/browse/IA-2398)

## Self proofreading checklist

- [x] Did I use eslint and black formatters
- [x] Is my code clear enough and well documented
- [x] New translations have been added or updated if new strings have been introduced in the frontend
- [x] Are there enough tests

## Changes

- Add a `Team` filter on users page
- Fix some typos in the doc and keep `README.md` and `docs/pages/dev/how_to/setup_dev_env/setup_dev_env.md` in sync

## Print screen / video

screen

[IA-2398]: https://bluesquare.atlassian.net/browse/IA-2398?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by beygorghor** [comment]
@cypress

File: hat/assets/js/apps/Iaso/domains/teams/components/CreateEditTeam.tsx [hunk 1]
```diff
@@ -89,13 +89,6 @@ export const CreateEditTeam: FunctionComponent<Props> = ({
         useGetProjectsDropDown();
     const { data: profliesDropdown, isFetching: isFetchingProfiles } =
         useGetProfilesDropdown();
-    const { data: teamsDropdown = [], isFetching: isFetchingTeams } =
-        useGetTeamsDropdown(
-            {
-                project,
-            },
-            id,
-        );
```
Comment (beygorghor) [inline hat/assets/js/apps/Iaso/domains/teams/components/CreateEditTeam.tsx]: This is still used to compute `availableChildren`

**Review by beygorghor — CHANGES_REQUESTED** [review]
You should probably adapt cypress tests to include this new filter. Also i think you removed used code. Otherwise great job 😉

File: hat/assets/js/apps/Iaso/domains/teams/components/CreateEditTeam.tsx (hunk 1 above)
Comment (kemar) [reply hat/assets/js/apps/Iaso/domains/teams/components/CreateEditTeam.tsx]: Fixed.

File: hat/assets/js/apps/Iaso/domains/teams/components/CreateEditTeam.tsx [hunk 2]
```diff
@@ -87,15 +87,15 @@ export const CreateEditTeam: FunctionComponent<Props> = ({
...
-    const { data: profliesDropdown, isFetching: isFetchingProfiles } =
-        useGetProfilesDropdown();
     const { data: teamsDropdown = [], isFetching: isFetchingTeams } =
         useGetTeamsDropdown(
             {
                 project,
             },
             id,
         );
+    const { data: profliesDropdown, isFetching: isFetchingProfiles } =
```
Comment (beygorghor) [inline hat/assets/js/apps/Iaso/domains/teams/components/CreateEditTeam.tsx]: `proflies` lol

File: hat/assets/js/cypress/README.md [hunk 3]
```diff
@@ -5,12 +5,22 @@ Development environment - Cypress e2e tests
...
+Prerequisites
+-------------
+
+Before you can run Cypress, you must have a version of Node.js and npm that matches the current Dockerfile:
+
+- Install [nvm](https://github.com/nvm-sh/nvm) locally
+- `nvm install 14.17`
+- `nvm use 14`
+- `npm install -g npm@8.5`
+
```
Comment (beygorghor) [inline hat/assets/js/cypress/README.md:17]: ❤️

**Review by beygorghor — APPROVED** [review]
One more typo :trollface:

**Pull request merged**